File: tests/fixtures/patterns-positive/chain_of_responsibility/mutB.py
# POSITIVO (mutación B: cadena de responsabilidad aplanada; pila de guardas repetida).
class RequestPipeline:
    def handle_a(self, request, validator, logger):
        if validator:
            validator.check(request)
        if logger:
            logger.log(request)

    def handle_b(self, request, validator, logger):
        if validator:
            validator.check(request)
        if logger:
            logger.log(request)

File: tests/fixtures/patterns-positive/chain_of_responsibility/mutB.ts
// POSITIVO (mutación B: cadena de responsabilidad aplanada; pila de guardas repetida).
class RequestPipeline {
  handleA(request: Request, validator: Validator | null, logger: Logger | null): void {
    if (validator) {
      validator.check(request);
    }
    if (logger) {
      logger.log(request);
    }
  }
  handleB(request: Request, validator: Validator | null, logger: Logger | null): void {
    if (validator) {
      validator.check(request);
    }
    if (logger) {
      logger.log(request);
    }
  }
}
